tests/alerts/plugins/test_dashboard_geomodel.py

```diff
@@ -50,12 +50,23 @@ def test_geoip_message_good(self):
         assert result_db_entry['description'] == 'This alert is created based on geo ip information about the last login of a user.'
         assert result_db_entry['duplicate'] is True
         assert result_db_entry['risk'] == 'high'
-        assert result_db_entry['summary'] == 'Did you recently login from Diamond Bar, United States?'
+        assert result_db_entry['summary'] == 'Did you recently login from Diamond Bar, United States (1.2.3.4)?'
         assert result_db_entry['url'] == 'https://www.mozilla.org'
         assert result_db_entry['url_title'] == 'Get Help'
         assert result_db_entry['user_id'] == 'ttesterson'
 
-    def test_geoip_message_bad(self):
+    def test_unknown_city_message(self):
+        message_dict = self.good_message_dict
+        message_dict['details']['locality_details']['city'] = 'UNKNOWN'
+        assert self.test_result_record is None
+        result_message = self.plugin.onMessage(message_dict)
+        assert result_message == self.good_message_dict
+        assert self.test_connect_called is True
+        result_db_entry = self.test_result_record
+        assert type(result_db_entry['alert_code']) is str
+        assert result_db_entry['summary'] == 'Did you recently login from United States (1.2.3.4)?'
```

Comment:
I also modified the summary so that we stuff the IP address at the end of the message. This resolves https://github.com/mozilla/MozDef/issues/536